{
  "uid" : "95c8fb53084e44b4",
  "name" : "test_returns_true",
  "fullName" : "tests.brownie.token.tests.test_transferFrom#test_returns_true",
  "historyId" : "592db1f5b525cbb8a4e298823c44ebef",
  "time" : {
    "start" : 1642034869627,
    "stop" : 1642034873713,
    "duration" : 4086
  },
  "status" : "broken",
  "statusMessage" : "brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'",
  "statusTrace" : "accounts = <brownie.network.account.Accounts object at 0x7f991e92af70>\ntoken = <Token Contract '0x44B0A2184e8C42B5D91aB94C9f5Afc20F776f846'>\n\n    def test_returns_true(accounts, token):\n        amount = token.balanceOf(accounts[0])\n        token.approve(accounts[1], amount, {'from': accounts[0]})\n        tx = token.transferFrom(accounts[0], accounts[2], amount, {'from': accounts[1]})\n    \n>       assert tx.return_value is True\n\ntests/brownie/token/tests/test_transferFrom.py:71: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/transaction.py:53: in wrapper\n    raise exc\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/transaction.py:49: in wrapper\n    return fn(self)\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/transaction.py:254: in return_value\n    self._get_trace()\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <Transaction '\u001B[0;m0x41d9d3b03f8289dfdb57cc55316f5c7ac591820b9dd516f5e6868d9beda2e36a\u001B[0;m'>\n\n    def _get_trace(self) -> None:\n        \"\"\"Retrieves the stack trace via debug_traceTransaction and finds the\n        return value, revert message and event logs in the trace.\n        \"\"\"\n    \n        # check if trace has already been retrieved, or the tx warrants it\n        if self._raw_trace is not None:\n            return\n        self._raw_trace = []\n        if self.input == \"0x\" and self.gas_used == 21000:\n            self._modified_state = False\n            self._trace = []\n            return\n    \n        if not web3.supports_traces:\n            raise RPCRequestError(\"Node client does not support `debug_traceTransaction`\")\n        try:\n            trace = web3.provider.make_request(  # type: ignore\n                \"debug_traceTransaction\", (self.txid, {\"disableStorage\": CONFIG.mode != \"console\"})\n            )\n        except (requests.exceptions.Timeout, requests.exceptions.ConnectionError) as e:\n            msg = f\"Encountered a {type(e).__name__} while requesting \"\n            msg += \"`debug_traceTransaction`. The local RPC client has likely crashed.\"\n            if CONFIG.argv[\"coverage\"]:\n                msg += \" If the error persists, add the `skip_coverage` marker to this test.\"\n            raise RPCRequestError(msg) from None\n    \n        if \"error\" in trace:\n            self._modified_state = None\n            self._trace_exc = RPCRequestError(trace[\"error\"][\"message\"])\n>           raise self._trace_exc\nE           brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'\n\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/transaction.py:668: RPCRequestError",
  "flaky" : false,
  "newFailed" : false,
  "newBroken" : false,
  "newPassed" : false,
  "retriesCount" : 0,
  "retriesStatusChange" : false,
  "beforeStages" : [ {
    "name" : "configure_neon_network",
    "time" : {
      "start" : 1642034771706,
      "stop" : 1642034771706,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "brownie_project",
    "time" : {
      "start" : 1642034773193,
      "stop" : 1642034789605,
      "duration" : 16412
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "setup_env",
    "time" : {
      "start" : 1642034793303,
      "stop" : 1642034793303,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "accounts",
    "time" : {
      "start" : 1642034793303,
      "stop" : 1642034793572,
      "duration" : 269
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "token",
    "time" : {
      "start" : 1642034839866,
      "stop" : 1642034842843,
      "duration" : 2977
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "status" : "broken",
    "statusMessage" : "brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'",
    "statusTrace" : "accounts = <brownie.network.account.Accounts object at 0x7f991e92af70>\ntoken = <Token Contract '0x44B0A2184e8C42B5D91aB94C9f5Afc20F776f846'>\n\n    def test_returns_true(accounts, token):\n        amount = token.balanceOf(accounts[0])\n        token.approve(accounts[1], amount, {'from': accounts[0]})\n        tx = token.transferFrom(accounts[0], accounts[2], amount, {'from': accounts[1]})\n    \n>       assert tx.return_value is True\n\ntests/brownie/token/tests/test_transferFrom.py:71: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/transaction.py:53: in wrapper\n    raise exc\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/transaction.py:49: in wrapper\n    return fn(self)\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/transaction.py:254: in return_value\n    self._get_trace()\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <Transaction '\u001B[0;m0x41d9d3b03f8289dfdb57cc55316f5c7ac591820b9dd516f5e6868d9beda2e36a\u001B[0;m'>\n\n    def _get_trace(self) -> None:\n        \"\"\"Retrieves the stack trace via debug_traceTransaction and finds the\n        return value, revert message and event logs in the trace.\n        \"\"\"\n    \n        # check if trace has already been retrieved, or the tx warrants it\n        if self._raw_trace is not None:\n            return\n        self._raw_trace = []\n        if self.input == \"0x\" and self.gas_used == 21000:\n            self._modified_state = False\n            self._trace = []\n            return\n    \n        if not web3.supports_traces:\n            raise RPCRequestError(\"Node client does not support `debug_traceTransaction`\")\n        try:\n            trace = web3.provider.make_request(  # type: ignore\n                \"debug_traceTransaction\", (self.txid, {\"disableStorage\": CONFIG.mode != \"console\"})\n            )\n        except (requests.exceptions.Timeout, requests.exceptions.ConnectionError) as e:\n            msg = f\"Encountered a {type(e).__name__} while requesting \"\n            msg += \"`debug_traceTransaction`. The local RPC client has likely crashed.\"\n            if CONFIG.argv[\"coverage\"]:\n                msg += \" If the error persists, add the `skip_coverage` marker to this test.\"\n            raise RPCRequestError(msg) from None\n    \n        if \"error\" in trace:\n            self._modified_state = None\n            self._trace_exc = RPCRequestError(trace[\"error\"][\"message\"])\n>           raise self._trace_exc\nE           brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'\n\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/transaction.py:668: RPCRequestError",
    "steps" : [ ],
    "attachments" : [ {
      "uid" : "b062dcfbbe4353b8",
      "name" : "stdout",
      "source" : "b062dcfbbe4353b8.txt",
      "type" : "text/plain",
      "size" : 622
    } ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 1,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "accounts::0",
    "time" : {
      "start" : 1642035057855,
      "stop" : 1642035057855,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "feature",
    "value" : "Brownie"
  }, {
    "name" : "parentSuite",
    "value" : "tests.brownie.token.tests"
  }, {
    "name" : "suite",
    "value" : "test_transferFrom"
  }, {
    "name" : "host",
    "value" : "fv-az135-192"
  }, {
    "name" : "thread",
    "value" : "3635-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.brownie.token.tests.test_transferFrom"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Not supported",
      "messageRegex" : ".*'EthereumModel' object has no attribute .*",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 92,
        "skipped" : 0,
        "passed" : 0,
        "unknown" : 0,
        "total" : 92
      },
      "items" : [ {
        "uid" : "f65cc01f4ce67b47",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/479//#testresult/f65cc01f4ce67b47",
        "status" : "broken",
        "statusDetails" : "brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'",
        "time" : {
          "start" : 1642033030983,
          "stop" : 1642033034657,
          "duration" : 3674
        }
      }, {
        "uid" : "aff55edcea55966e",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/478//#testresult/aff55edcea55966e",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: http://proxy.night.stand.neontest.xyz/solana",
        "time" : {
          "start" : 1642032074846,
          "stop" : 1642032074846,
          "duration" : 0
        }
      }, {
        "uid" : "1349665d680cac51",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/477//#testresult/1349665d680cac51",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642031189490,
          "stop" : 1642031189490,
          "duration" : 0
        }
      }, {
        "uid" : "5e6dabab7042c58f",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/476//#testresult/5e6dabab7042c58f",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642028351115,
          "stop" : 1642028351115,
          "duration" : 0
        }
      }, {
        "uid" : "c7659b9886a9a1fc",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/474//#testresult/c7659b9886a9a1fc",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642025673250,
          "stop" : 1642025673250,
          "duration" : 0
        }
      }, {
        "uid" : "79ff18aa79302c1",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/472//#testresult/79ff18aa79302c1",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642018238899,
          "stop" : 1642018238899,
          "duration" : 0
        }
      }, {
        "uid" : "30cceff25e8932b5",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/471//#testresult/30cceff25e8932b5",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642015604930,
          "stop" : 1642015604930,
          "duration" : 0
        }
      }, {
        "uid" : "ece2a52aa1f13de5",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/470//#testresult/ece2a52aa1f13de5",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642014231044,
          "stop" : 1642014231044,
          "duration" : 0
        }
      }, {
        "uid" : "c56ac2fa13b22e1d",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/469//#testresult/c56ac2fa13b22e1d",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642012912513,
          "stop" : 1642012912513,
          "duration" : 0
        }
      }, {
        "uid" : "ecd1f0d1e89e97a",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/468//#testresult/ecd1f0d1e89e97a",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642011423283,
          "stop" : 1642011423283,
          "duration" : 0
        }
      }, {
        "uid" : "7ef26b3e410a0e85",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/466//#testresult/7ef26b3e410a0e85",
        "status" : "broken",
        "statusDetails" : "brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'",
        "time" : {
          "start" : 1641998016235,
          "stop" : 1641998020703,
          "duration" : 4468
        }
      }, {
        "uid" : "3f99228908694a56",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/465//#testresult/3f99228908694a56",
        "status" : "broken",
        "statusDetails" : "brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'",
        "time" : {
          "start" : 1641993777639,
          "stop" : 1641993782151,
          "duration" : 4512
        }
      }, {
        "uid" : "b99cab39bf05a41a",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/464//#testresult/b99cab39bf05a41a",
        "status" : "broken",
        "statusDetails" : "brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'",
        "time" : {
          "start" : 1641991033532,
          "stop" : 1641991037409,
          "duration" : 3877
        }
      }, {
        "uid" : "1bb40621081fa671",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/463//#testresult/1bb40621081fa671",
        "status" : "broken",
        "statusDetails" : "brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'",
        "time" : {
          "start" : 1641986382135,
          "stop" : 1641986384938,
          "duration" : 2803
        }
      }, {
        "uid" : "c0f101e41d9d2b7d",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/461//#testresult/c0f101e41d9d2b7d",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1895fe5e80>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641982478830,
          "stop" : 1641982478830,
          "duration" : 0
        }
      }, {
        "uid" : "e8483428d0518248",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/460//#testresult/e8483428d0518248",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f189460db20>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641980267870,
          "stop" : 1641980267870,
          "duration" : 0
        }
      }, {
        "uid" : "747eeafea5fdb37a",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/459//#testresult/747eeafea5fdb37a",
        "status" : "broken",
        "statusDetails" : "brownie.exceptions.RPCRequestError: 'EthereumModel' object has no attribute 'debug_traceTransaction'",
        "time" : {
          "start" : 1641975850091,
          "stop" : 1641975853757,
          "duration" : 3666
        }
      }, {
        "uid" : "a6bda6f4ab29e595",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/458//#testresult/a6bda6f4ab29e595",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f189a842400>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641974827929,
          "stop" : 1641974827929,
          "duration" : 0
        }
      }, {
        "uid" : "a59833fc5de53206",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/457//#testresult/a59833fc5de53206",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1896042940>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641974206501,
          "stop" : 1641974206501,
          "duration" : 0
        }
      }, {
        "uid" : "1bfadc0f4075ac40",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/456//#testresult/1bfadc0f4075ac40",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.ReadTimeout: HTTPConnectionPool(host='proxy.night.stand.neontest.xyz', port=80): Read timed out. (read timeout=30)",
        "time" : {
          "start" : 1641958011607,
          "stop" : 1641958011607,
          "duration" : 0
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "95c8fb53084e44b4.json",
  "parameterValues" : [ ]
}